app/apps/workflow/models.py

```diff
@@ -132,76 +138,83 @@ def wait_for_workflows_and_send_message(message):
 
             # tests if all workflows reached thit point
             if len(workflows_completed) == all_workflows.count() and main_workflow:
+                from .tasks import accept_message_for_workflow
 
                 # pick up all summons and pass them on to the main workflow
                 all_summons = [
                     d.get("summon_id")
                     for d in all_workflows.values_list("data", flat=True)
                     if d.get("summon_id")
                 ]
-                messages = main_workflow.data.get("messages", [])
+                extra_data = {
+                    "all_summons": all_summons,
+                }
 
-                # add message to main workflow, this will tell only the main workflow to go on
-                messages.append(message)
-                main_workflow.data.update(
-                    {
-                        "messages": messages,
-                        "all_summons": all_summons,
-                    }
-                )
-                main_workflow.save(update_fields=["data"])
-                main_workflow.update_workflow()
+                accept_message_for_workflow.delay(main_workflow.id, message, extra_data)
 
                 # delete others
-                other_workflows = all_workflows.exclude(id=main_workflow.id)
-                other_workflows.delete()
+                # other_workflows = all_workflows.exclude(id=main_workflow.id)
+                # other_workflows.delete()
```

Comment:
Why is this commented out? Can it be removed or is it an accidental change?

Reply:
Comments are on the way for that